### ISSUE TITLE
Minification breaks userapp-angular

### ISSUE THREAD

**Issue by johanbuys** [userapp-io/userapp-angular#11]
Hi, 

After a lengthy debugging session on my own app I was able to isolate a problem I was having to the UsesApp-Angular module.

To try and proof that it was not my code I used the example app code from this repo and the same issue occurred. 

**The Issue: When you minnify and combine js sources angular moans about "Unknown Provider"**

**My Setup:**

Using _Gulp_ I minify and combine the example  using gulp-useref and gulp-uglify then start the app using gulp-connect. The App fails with the standard angular provider error:

> Uncaught Error: [$injector:unpr] Unknown provider: eProvider <- e
> http://errors.angularjs.org/1.2.3/$injector/unpr?p0=eProvider%20%3C-%20e 

From my experience this is mostly caused by the injector, which is unable to find the module to inject, when a minifier changes variable names and thus changes the name the injector will search for, usually this is easily solved by using the array syntax when defining controllers, services and the like ie

```
  angular.controller('contr',['$scope',function($scope){}]);
```

Unfortunately up to now I am still unable to find the erroneous module.
Been spending a few days on this now. Hope some one can help with this. I will try and continue to find out what the issue might be.

**Comment by madecostore** [comment]
Hi!
Same issue for me with yeoman grunt build.

**Comment by ghost** [comment]
Same issue here with gulp. Is this going to be fixed?

**Comment by luigiw** [comment]
I had the same problem, fixed it with below code.

myApp.run(['user',
  (user)->
    user.init(
      appId: '12345'
    )
])

**Comment by viniciusmelquiades** [comment]
I found the line where the error occurs, after minifying. I'll test it a little in the project I was using it and create a PR (today) if it nothing breaks down the line.

**Comment by jgangemi** [comment]
a year later and crickets

**Comment by viniciussalvati** [comment]
@jgangemi my pull request fixes this. All they needed to do was merge it. You can use my fork if you want.

**Comment by jgangemi** [comment]
yeah - that's what i'm about to do. i'm just annoyed that they can't integrate this into the main line, it's been over a year.

**Comment by viniciussalvati** [comment]
The lastest update to the repository was over a year ago. I wonder if they're interacting on the other issues.

Tbh I had one (horrible) experience using UserApp and stopped using it since then, and don't plan on returning.

**Comment by jgangemi** [comment]
what did you do for an alternative?

**Comment by viniciussalvati** [comment]
@jgangemi We started using Auth0 where I work. At least they seem to keep their plugins updated and working

**Comment by jgangemi** [comment]
@viniciusmelquiades thanks! i came across them yesterday as well and will be dropping userapp, especially after i kept getting constant errors on their website last night trying to do some things.

**Comment by viniciussalvati** [comment]
@jgangemi they have some very cool features, even for free users. I'm glad I could help
